google/cloud/spanner_v1/instance.py

```diff
@@ -125,6 +128,11 @@ def __init__(
         self.configuration_name = configuration_name
         self.node_count = node_count
         self.processing_units = processing_units
+        if node_count is None and processing_units is None:
+            self.node_count = DEFAULT_NODE_COUNT
+        if node_count is not None and processing_units is not None:
+            raise InvalidArgument(
+                'Only one of node count and processing units can be set.')
```

Comment:
nit: put the validation check before setting the default

If the validation fails then the default isn't needed.